app/scripts/modules/core/instance/instance.write.service.js

```diff
@@ -6,8 +6,72 @@ module.exports = angular
   .module('spinnaker.core.instance.write.service', [
     require('../task/taskExecutor.js'),
     require('../serverGroup/serverGroup.read.service.js'),
+    require('../utils/lodash.js'),
   ])
-  .factory('instanceWriter', function (taskExecutor, serverGroupReader) {
+  .factory('instanceWriter', function (taskExecutor, serverGroupReader, _) {
+
+    function buildMultiInstanceJob(instanceGroups, type) {
+      let jobsByZone = instanceGroups
+        .filter((instanceGroup) => instanceGroup.instances.length > 0)
+        .map((instanceGroup) => {
+          /*
+          // TODO: ttomsu - when zone and launchTimes aren't required, skip the groupBy stuff below and just use this
+           return {
+             type: type,
+             cloudProvider: instanceGroup.cloudProvider,
+             providerType: instanceGroup.cloudProvider,
+             instanceIds: instances.map((instance) => instance.id),
+             credentials: instanceGroup.account,
+             region: instanceGroup.region,
+             serverGroupName: instanceGroup.serverGroup
+           };
+           */
```

Comment:
Yeah, it's a single task with multiple `job`s - at least one for each server group. The grouping is needed as long as `zone` is required, since a server group will typically contain instances from multiple zones, but the task signature in orca assumes all instances are in the same zone - until your PR around terminating instances (https://github.com/spinnaker/orca/pull/689) is merged, at least, that's the case. 

Once it's merged, we really don't need the `_.groupBy` at all, so you can disregard the original comment.